pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -93,32 +105,97 @@ func (c *Controller) Run(stop <-chan struct{}) {
 		go c.periodicWorkloadEntryCleanup(stop)
 		go c.cleanupQueue.Run(stop)
 	}
+
+	for i := 0; i < 5; i++ {
+		go wait.Until(c.worker, time.Second, stop)
+	}
+}
+
+func (c *Controller) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+type WorkType int
+
+const (
+	Register WorkType = iota
+	Unregister
+)
+
+type workItem struct {
+	entryName  string
+	workType   WorkType
+	proxy      *model.Proxy
+	conTime    time.Time
+	disConTime time.Time
+}
+
+func (c *Controller) processNextWorkItem() bool {
+	item, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(item)
+
+	workItem, ok := item.(*workItem)
+	if !ok {
+		return true
+	}
+
+	switch workItem.workType {
+	case Register:
+		err := c.registerWorkload(workItem.entryName, workItem.proxy, workItem.conTime)
+		c.handleErr(err, item)
```

Comment:
If handleErr requeues `register` but an `unregister` is already on the queue, I think we'd end up in a wrong state with a WorkloadEntry that shouldn't exist.

Reply:
Right,  will figure out how to handle this

Reply:
Reverted retrying registerWorkload, leaves it to grpc reconnect to handle the error